scripts/rosversion

```diff
@@ -68,18 +74,31 @@ def get_distro_name_from_roscore():
         traceback.print_exc()
 
 
-if len(sys.argv) == 2:
+def get_version_printer(nonewline=False):
+    """
+    @return: A method object to print the version string.
+    """
+    if nonewline:
+        return sys.stderr.write
+    else:
+        return print
+
+if len(sys.argv) >= 2:
+    version_printer = print
     stack_name = sys.argv[1]
 
+    if len(sys.argv) == 3 and sys.argv[2] == "-n":
```

Comment:
Instead of starting to manually handle new arguments with custom logic please use `argparse` for it.

Reply:
I agree to use `argparse` or something organized argument parser. But because 2 usecases of `rosversion` where both takes a positional args, one of which starts with hyphen so that `argparse` automatically recognizes it as an optional, I can't think of how to implement using `argparse`. I'll keep thinking but if you can share your idea if any I'd appreciate.

```
rosversion <package/stack> 
```
or
```
rosversion -d                                                                                                                                                 
```

Reply:
[Asked on SO](https://stackoverflow.com/questions/48798761/positional-arg-starts-with-hyphen-in-argparse).

Reply:
```
parser = argparse.ArgumentParser()
group = parser.add_mutually_exclusive_group()
group.add_argument('package', nargs='?')
group.add_argument('-d', action='store_true')
```